### PR TITLE
chore: add unit tests for authz related functions

### PR DESCRIPTION
**Pull request by mevan-karu**
Related to https://github.com/openchoreo/openchoreo/issues/2996

This pull request significantly expands test coverage for the Casbin authorization logic, focusing on both the policy decision point (`pdp.go`) and policy administration point (`enforcer.go`). The new tests cover error handling, edge cases, and validation logic, ensuring more robust behavior for policy evaluation, resource/role listing, and input validation.

The most important changes include:

**Expanded test coverage for policy evaluation and error handling:**

* Added tests for `Evaluate`, `BatchEvaluate`, and `GetSubjectProfile` methods to cover nil requests, context cancellation, missing or malformed subject context, malformed policies, and out-of-scope policies. This ensures the system handles invalid input and edge cases gracefully.

* Added tests for the `check` method to verify correct deny decisions and error returns when subject context is missing or malformed.

**Enforcer instantiation and configuration:**

* Added tests for `NewEnforcer` to cover error cases (such as nil Kubernetes client) and cache configuration, including default TTL handling.

**Validation and helper function testing:**

* Added tests for context and resource matching helpers (`ctxMatch`, `resourceMatchWrapper`, `ctxMatchWrapper`) to verify correct match logic and error handling for invalid arguments.

* Added a test for batch evaluation request validation, ensuring that requests with missing actions are correctly rejected.

**Policy administration (role and binding management):**

* Added comprehensive tests for listing cluster and namespaced roles and role bindings, including pagination (`limit` and `cursor`) scenarios and not-found errors for update operations. This improves confidence in the policy administration API.

**Minor improvements:**

* Included additional imports (`io`, `time`) required for new tests.

### CODE REVIEW

File: internal/authz/casbin/helpers_test.go [hunk 1]
```diff
@@ -1110,3 +1110,118 @@ func sliceContainsSameElements(a, b []string) bool {
 	}
 	return true
 }
+
+func TestCtxMatch(t *testing.T) {
```
Comment (binoyPeries) [inline internal/authz/casbin/helpers_test.go]: There's no logic for this fn currently, this will be updated with the upcoming ABAC impl. I think we this is not requried until then. WDYT?

File: internal/authz/casbin/helpers_test.go [hunk 2]
```diff
@@ -1110,3 +1110,118 @@ func sliceContainsSameElements(a, b []string) bool {
...
+			requestCtx: "env:staging",
+			policyCtx:  "env:prod",
+			want:       false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := ctxMatch(tt.requestCtx, tt.policyCtx)
+			if got != tt.want {
+				t.Errorf("ctxMatch(%q, %q) = %v, want %v", tt.requestCtx, tt.policyCtx, got, tt.want)
```
Comment (binoyPeries) [inline internal/authz/casbin/helpers_test.go]: we use testify for assertions. refer existing test cases

File: internal/authz/casbin/k8s_watcher_test.go [hunk 3]
```diff
@@ -1735,3 +1736,925 @@ func TestAuthzInformerHandler_HandleDeleteClusterBinding_Scoped(t *testing.T) {
...
+		ObjectMeta: metav1.ObjectMeta{Name: "on-upd-role", Namespace: "acme", Generation: 1},
+		Spec:       authzv1alpha1.AuthzRoleSpec{Actions: []string{"component:view"}},
+	}
+	newR := &authzv1alpha1.AuthzRole{
+		ObjectMeta: metav1.ObjectMeta{Name: "on-upd-role", Namespace: "acme", Generation: 2},
+		Spec:       authzv1alpha1.AuthzRoleSpec{Actions: []string{"component:create"}},
+	}
+	// seed the old policy first so remove succeeds cleanly
+	_, _ = h.enforcer.AddGroupingPolicy("on-upd-role", "component:view", "acme")
+	h.OnUpdate(old, newR) // should not panic
```
Comment (binoyPeries) [inline internal/authz/casbin/k8s_watcher_test.go]: I'm a bit unclear on what we are testing here :)

File: internal/authz/casbin/pdp_test.go [hunk 4]
```diff
@@ -1398,3 +1400,263 @@ func TestCasbinEnforcer_Evaluate_ScopedClusterBinding(t *testing.T) {
...
+func TestNewEnforcer_WithCacheDefaultTTL(t *testing.T) {
+	logger := slog.New(slog.NewTextHandler(io.Discard, nil))
+	ce := setupTestEnforcer(t)
+	// CacheTTL=0 should use the default (5m)
+	_, err := NewEnforcer(context.Background(), CasbinConfig{
+		K8sClient:    ce.k8sClient,
+		CacheEnabled: true,
+		CacheTTL:     0,
+	}, logger)
+	if err != nil {
```
Comment (binoyPeries) [inline internal/authz/casbin/pdp_test.go]: Shouldn't we test here whether enforce struct is populated properly?

File: internal/authz/casbin/helpers_test.go (hunk 1 above)
Comment (mevan-karu) [reply internal/authz/casbin/helpers_test.go]: +1. Will remove these tests for now.

Comment (mevan-karu) [reply internal/authz/casbin/helpers_test.go]: Updated.

File: internal/authz/casbin/helpers_test.go (hunk 2 above)
Comment (mevan-karu) [reply internal/authz/casbin/helpers_test.go]: Updated the new tests.

File: internal/authz/casbin/k8s_watcher_test.go (hunk 3 above)
Comment (mevan-karu) [reply internal/authz/casbin/k8s_watcher_test.go]: This was added to verify the proper CRD updates are happening to the controller runtime-cache. Updated the test with some assertions to increase the explainability of the test case.

File: internal/authz/casbin/pdp_test.go (hunk 4 above)
Comment (mevan-karu) [reply internal/authz/casbin/pdp_test.go]: Fixed.
